autopush/db.py

```diff
@@ -432,45 +445,70 @@ def generate_last_connect_values(date):
             yield int(val)
 
 
-def list_tables(client=g_client):
-    """Return a list of the names of all DynamoDB tables."""
-    start_table = None
-    while True:
-        if start_table:  # pragma nocover
-            result = client.list_tables(ExclusiveStartTableName=start_table)
-        else:
-            result = client.list_tables()
-        for table in result.get('TableNames', []):
-            yield table
-        start_table = result.get('LastEvaluatedTableName', None)
-        if not start_table:
-            break
+def table_exists(tablename, boto_resource=None):
+    # type: (str, DynamoDBResource) -> bool
+    """Determine if the specified Table exists"""
+    try:
+        return boto_resource.Table(tablename).table_status in [
+            'CREATING', 'UPDATING', 'ACTIVE']
+    except ClientError:
+        return False
 
 
-def table_exists(tablename, client=None):
-    """Determine if the specified Table exists"""
-    if not client:
-        client = g_client
-    return tablename in list_tables(client)
+class DynamoDBResource(threading.local):
+    def __init__(self, **kwargs):
+        conf = kwargs
+        if not conf.get("endpoint_url"):
```

Comment:
this is now aws_ddb_endpoint. we don't need to handle AWS_LOCAL_DYNAMODB env here any longer since configargparse is now doing it for us